### PR TITLE
ikev1: add ikev1 parser - v8

### PR DESCRIPTION
**Pull request by fhonza**
Previous PR: #5375 

Describe changes:
- rearranged commits
- fix clang errors
- cleanup function for AppLayerProtoDetectAliases added

#suricata-verify-pr: [334](https://github.com/OISF/suricata-verify/pull/334)

### CODE REVIEW

**Comment by catenacyber** [comment]
There still seems to be a CI failure for this PR.
@fhonza did you try `make check` locally ?

**Comment by fhonza** [comment]
> There still seems to be a CI failure for this PR.
> @fhonza did you try `make check` locally ?

sry haven't run this locally, will check it!

File: doc/userguide/rules/ike-keywords.rst [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+``ikev1.key_exchange_payload`` is a 'sticky buffer'.
+
+``ikev1.key_exchange_payload`` can be used as ``fast_pattern``.
+
+
+ikev1.key_exchange_payload_length
+---------------------------------
+
+Match against the length of the public key exchange payload (e.g. Diffie-Hellman) of the Server or Client.
```
Comment (jlucovsky) [inline doc/userguide/rules/ike-keywords.rst:102]: nit: neither server nor client should be capitalizd.

**Comment by jlucovsky** [comment]
> #suricata-verify-pr: [334](https://github.com/OISF/suricata-verify/pull/334)

Remove the comment character and use
`suricata-verify-pr: 334` (without the link)

File: src/app-layer-ike.c [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        0x01, 0x00, 0x00, 0x80, 0x01, 0x00, 0x07, 0x80, 0x0e, 0x00, 0x80, 0x80, 0x02, 0x00, 0x02,
+        0x80, 0x04, 0x00, 0x02, 0x80, 0x03, 0x00, 0x01, 0x80, 0x0b, 0x00, 0x01, 0x00, 0x0c, 0x00,
+        0x04, 0x00, 0x01, 0x51, 0x80, 0x0d, 0x00, 0x00, 0x14, 0x4a, 0x13, 0x1c, 0x81, 0x07, 0x03,
+        0x58, 0x45, 0x5c, 0x57, 0x28, 0xf2, 0x0e, 0x95, 0x45, 0x2f, 0x0d, 0x00, 0x00, 0x14, 0x43,
+        0x9b, 0x59, 0xf8, 0xba, 0x67, 0x6c, 0x4c, 0x77, 0x37, 0xae, 0x22, 0xea, 0xb8, 0xf5, 0x82,
+        0x0d, 0x00, 0x00, 0x14, 0x7d, 0x94, 0x19, 0xa6, 0x53, 0x10, 0xca, 0x6f, 0x2c, 0x17, 0x9d,
+        0x92, 0x15, 0x52, 0x9d, 0x56, 0x00, 0x00, 0x00, 0x14, 0x90, 0xcb, 0x80, 0x91, 0x3e, 0xbb,
+        0x69, 0x6e, 0x08, 0x63, 0x81, 0xb5, 0xec, 0x42, 0x7b, 0x1f };
+
+    FLOWLOCK_WRLOCK(&f);
```
Comment (jlucovsky) [inline src/app-layer-ike.c:83]: The locking is not needed for the unit tests.

File: src/detect-ike-chosen-sa.c [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+ *
+ *  \retval 1 on success
+ *  \retval 0 on failure
+ */
+static int IKEChosenSaParserTest(void)
+{
+    DetectIkeChosenSaData *de = NULL;
+    de = DetectIkeChosenSaParse("hash_algorithm=2");
+
+    FAIL_IF_NULL(de);
```
Comment (jlucovsky) [inline src/detect-ike-chosen-sa.c:268]: Suggest that you validate the type and value inside of `de`

File: src/detect-ike-exch-type.c [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        void *state, void *txv, const Signature *s, const SigMatchCtx *ctx)
+{
+    SCEnter();
+
+    uint8_t exch_type;
+    if (!rs_ike_state_get_exch_type(txv, &exch_type))
+        SCReturnInt(0);
+
+    const DetectU8Data *du8 = (const DetectU8Data *)ctx;
+    return DetectU8Match(exch_type, du8);
```
Comment (jlucovsky) [inline src/detect-ike-exch-type.c:108]: Use `SCReturnInt`

File: src/detect-ike-vendor.c [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+ *  \param det_ctx detection engine thread ctx
+ *  \param p packet to inspect
+ *  \param f flow to inspect
+ *  \param txv tx to inspect
+ *  \param pectx inspection context
+ */
+static void PrefilterTxIkeVendor(DetectEngineThreadCtx *det_ctx, const void *pectx, Packet *p,
+        Flow *f, void *txv, const uint64_t idx, const uint8_t flags)
+{
+    SCEnter();
```
Comment (jlucovsky) [inline src/detect-ike-vendor.c:92]: Since you're using `SCEnter` .... use `SCReturn` on function exit.

File: src/output-json-ike.c [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+typedef struct LogIKEFileCtx_ {
+    LogFileCtx *file_ctx;
+    uint32_t flags;
+} LogIKEFileCtx;
+
+typedef struct LogIKELogThread_ {
+    LogIKEFileCtx *ikelog_ctx;
+    uint32_t count;
+    MemBuffer *buffer;
```
Comment (jlucovsky) [inline src/output-json-ike.c:62]: To support eve multi-threaded output, you'll need to add the `file_ctx` to this structure (see example from `output-json-ikev2.c`).

File: src/output-json-ike.c [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+        return TM_ECODE_FAILED;
+    }
+
+    thread->buffer = MemBufferCreateNew(JSON_OUTPUT_BUFFER_SIZE);
+    if (unlikely(thread->buffer == NULL)) {
+        SCFree(thread);
+        return TM_ECODE_FAILED;
+    }
+
+    thread->ikelog_ctx = ((OutputCtx *)initdata)->data;
```
Comment (jlucovsky) [inline src/output-json-ike.c:164]: You'll need to do this for multi-threaded eve output support (see `output-json-ikev2.c`
```
 thread->file_ctx = LogFileEnsureExists(thread->ikev2log_ctx->file_ctx, t->id);
    if (!thread->file_ctx) {
        goto error_exit;
    }
```

File: src/tests/fuzz/fuzz_sigpcap.c [hunk 8]
```diff
@@ -88,7 +88,7 @@ outputs:\n\
         - nfs\n\
         - smb\n\
         - tftp\n\
-        - ikev2\n\
+	- ike\n\
```
Comment (jlucovsky) [inline src/tests/fuzz/fuzz_sigpcap.c:91]: indentation

File: src/output-json-ike.c (hunk 6 above)
Comment (catenacyber) [reply src/output-json-ike.c:62]: This is a new feature, right ?

**Pull request closed without merging**